js/ui/grid_core/ui.grid_core.editing.js

```diff
@@ -654,21 +674,25 @@ const EditingController = modules.ViewController.inherit((function() {
             changes.forEach(change => {
                 const isInsert = change.type === DATA_EDIT_DATA_INSERT_TYPE;
 
-                if(!isInsert) {
+                if(!isInsert || this.needChangePageIndexToInsertRow()) {
```

Comment:
User can combine insertAfterKey and newRowPosition: 'bottom'. And in this case insertAfterKey should not be ignored. You can fix it in another PR.